src/Orleans/GrainDirectory/IGrainDirectory.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.GrainDirectory
+{
+
+    /// <summary>
+    /// Recursive distributed operations on grain directories.
+    /// Each operation may forward the request to a remote owner, increasing the hopcount.
+    /// </summary>
+    interface IGrainDirectory
```

Comment:
is this remotable (RPC-able) interface? I think it is.
If yes, our convention is to mark it as IAddressable, or ISystemTarget in this case.